packages/react/docs/guides/tech-stack.md

```diff
@@ -0,0 +1,283 @@
+
+are used through out the project when their compact style makes it much easier to read the code.
+
+```js
+// if only one argument you can even drop the parens
+setState((prevState) => !prevState);
+// vs
+setState(function (prevState) {
+  return !prevState;
+});
+```
+
+Information to keep in mind when dealing with Arrow functions:
+
+- basic syntax `() => {}`. The parens are always needed except...
+- When only one argument is used you can drop the parens. ex. `argument => { return argument++}`
+- Implicit return: Arrow functions can have either a "concise body" or the usual "block body". In a concise body, only an expression is specified, which becomes the implicit return value. ex. `argument => argument++`
+- Can not use `this` keyword like you would with traditional functions.
+- They do not have an `arguments` object. Both `arguments` and `this` reference the value of the enclosing lexical scope.
+
+There are crucial differences between arrow functions and traditional functional expressions so be aware and know [when to not use an arrow function](https://www.javascripttutorial.net/es6/when-you-should-not-use-arrow-functions/). With the above information in mind, I would lean towards using function delcaration for most instances and arrow functions when used as an argument in another function.
+
+When writing a functional component it is best practice to use a regular function declaration
+
+```js
+function MyComponent(props) {
+  return <div>Hello World</div>;
+}
+```
+
+but you will definitely see them defined using arrow functions both in our library and in the wild.
+
+```js
+const MyComponent = (props) => <div>Hello World</div>;
+```
+
+## [Template Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)
+
+are literals delimited with backticks (`), allowing embedded expressions called substitutions.
+
+```js
+const myString = `This is a string with ${variables}.`;
+```
+
+It can make strings with variable or expression iterpolation more readable and easier to write. Notice the `\n` is not needed because white space is not folded into one space as with string literals.
+
+```js
+const str1 = 'Fifteen is ' + (a + b) + ' and\nnot ' + (2 * a + b) + '.';
+const str2 = `Fifteen is ${a + b} and
+not ${2 * a + b}.`;
+```
+
+## [Tertiary conditional](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)
+
+is the only JavaScript operator that takes three operands: a condition followed by a question mark (?), then an expression to execute if the condition is truthy followed by a colon (:), and finally the expression to execute if the condition is falsy. This operator is frequently used as an alternative to an if...else statement.
+
+In react there will be times when you want to conditionally render an element. Like the content of an accordion.
+
+```js
+<AccordionItem>{isOpen ? <AccordionItemContent>Hey</AccordionItemContent> : null}</AccordionItem>
```

Comment:
While ternaries can make single line expressions very readable... I do find that they quickly become confusing when... 
a) they are nested
b) they include large multi-line responses on both the true and false expression

Reply:
Agreed, and there are cases where a traditional if statement may be necessary but for most cases I think the ternary is best practice over the use of `&&` to avoid the case of false rendering.

The example below would render `0`, unintentionally, if content was an empty array.

```js
{ content.length && <AccordionItemContent>content[i]</AccordionItemContent> }
```
I do think it is worth noting where the ternary can be abused and what to use in those cases.